manifests/install.pp

```diff
@@ -1,21 +1,10 @@
 # == Class gitlab::install
 #
 # This class is called from gitlab for install.
-#
-# [*package_name*]
-#   Default: 'gitlab-ce'
-#   The internal packaging system's name for the package
-#   This name will automatically be changed by the gitlab::edition parameter
-#   Can be overridden for the purposes of installing custom compiled version of gitlab-omnibus
-#
-# [*manage_package*]
-#   Default: true
-#   Should the GitLab package be managed?
-#
 class gitlab::install (
-  Optional[String] $package_name = undef,
+  $package_name = $gitlab::package_name,
   $package_ensure = $gitlab::package_ensure,
-  Boolean $manage_package = true,
+  $manage_package = $gitlab::manage_package,
```

Comment:
I think everything that is not a private class, should have datatypes for parameters. I dont see a reason for removing them here. Can you explain why you did that? Even if it is private I would not remove the datatype, I would just not spend time implementing them. If this class is supposed to be private we should add a assert_private() call at the top

Reply:
I moved the type validation for this param into `manifests/init.pp`.  I've never seen the `assert_private()` functionality, but if keeps the class from being called from outside the module, then yes, this would be something I will add.

If I _do_ add `assert_private()`, then I'm assuming leaving data type validation in `manifests/init.pp` _only_ is fine, or should we still put the validation in both places?

Reply:
I would still prefer it to keep the datatypes.

Reply:
I'm making development changes based on patterns i'm seeing in the rest of the VP modules, and I was just using the same standards I've seen elsewhere. I haven't seen any other examples where private classes have double verification. It's difficult for me to live up to your standards when they only apply to me.

https://github.com/voxpupuli/puppet-nginx/commit/206dc3d263b00ba9394aa6bf437f492431da51f3